lib/internal/errors.js

```diff
@@ -1361,6 +1361,9 @@ E('ERR_WASI_ALREADY_STARTED', 'WASI instance has already started', Error);
 E('ERR_WORKER_INVALID_EXEC_ARGV', (errors) =>
   `Initiated Worker with invalid execArgv flags: ${errors.join(', ')}`,
   Error);
+E('ERR_WORKER_INVALID_NODE_OPTIONS', (errors) =>
+  `Initiated Worker with invalid NODE_OPTIONS: ${errors.join(', ')}`,
+  Error);
```

Comment:
Tbh, I think this is a situation here we should just use `ERR_WORKER_INVALID_EXEC_ARGV`, as it is the same type of error that’s being reported

Reply:
Gave it another thought, imo `ERR_WORKER_INVALID_EXEC_ARGV` is way too specific and may mislead people into thinking that the error is elsewhere. I think it would be better to add `ERR_WORKER_INVALID_OPTIONS` and group both `ERR_WORKER_INVALID_EXEC_ARGV` and `ERR_WORKER_INVALID_NODE_OPTIONS` under it, though that may be breaking. Or just add and use `ERR_WORKER_INVALID_OPTIONS` here and then open a semver-major to change that in for `ERR_WORKER_INVALID_EXEC_ARGV`. wdyt?

Reply:
That’s a general problem with our error codes – they tend to be way too specific.

I’d be good with changing the code, but yes, that would be breaking and I’d like to avoid it in this PR.

I’d still be good with using `ERR_WORKER_INVALID_EXEC_ARGV` and then pointing out in the error message where the error came from (and, ideally, also in a property on the error object). We can always switch that to a different error code later.

(Fwiw, `ERR_WORKER_INVALID_OPTIONS` sounds like it refers to the options object passed to the `Worker` constructor…)

Reply:
> (Fwiw, ERR_WORKER_INVALID_OPTIONS sounds like it refers to the options object passed to the Worker constructor…)

That was the idea - part of the options passed is incorrect. tbh we can probably use `ERR_INVALID_OPT_VALUE` from https://github.com/nodejs/node/pull/31251 :smile:.

Then I'll use the `ERR_WORKER_INVALID_EXEC_ARGV` in this PR and then open another for the error change.